glambie/data/read_and_edit_bucket_csv_files.py

```diff
@@ -0,0 +1,343 @@
+
+    This function doesn't actually edit any files - instead it returns a record of which files should be edited and
+    why - which we want to save to retain full transparency for glambie participants.
+
+    Parameters
+    ----------
+    csv_file_path : str
+        Path to downloaded glambie csv file.
+    file_check_dataframe : pd.DataFrame
+        Dataframe containing results of checks for inconsistencies within the csv files compared to the expected glambie
+        standard.
+
+    Returns
+    -------
+    pd.DataFrame
+        Dataframe containing input dataframe updated with the results of the checks for one file.
+    """
+    print(os.path.basename(csv_file_path))
+    submission_data_frame = pd.read_csv(csv_file_path)
+
+    # First, check for any non equal end_dates and subsequent start_dates
+    start_dates = submission_data_frame.start_date.values
+    end_dates = submission_data_frame.end_date.values
+    start_dates = np.append(start_dates, np.nan)  # Add an extra element to the end of this list for check below
+
+    date_check_bool = True
+    nodata_check_bool = True
+
+    for ind, end_date in enumerate(end_dates):
+        # If end_date != start_date for any of the consecutive rows, set date_check_bool to False for this file - i.e.
+        # file does not pass test and will need to be edited
+        row_bool = end_date.__eq__(start_dates[ind + 1])
+        if not row_bool:
+            date_check_bool = False
+
+    # Next, check for any instances of extreme values of change - we are aware of instances where people have used
+    # e.g change = 9999. for rows where they don't have a measured change. They were unaware that they should just
+    # remove these rows instead of setting an arbitrary nodata value. Extreme value threshold needs to be different
+    # depending on the units used.
+    change_values = submission_data_frame.glacier_change_observed.values
+
+    if 'm' in submission_data_frame.unit.values[0]:  # check if units are m or mwe
+        nodata_check_bool = all(abs(i) < 100 for i in change_values)  # check that all changes in list are < +/-100
+    elif 'Gt' in submission_data_frame.unit.values[0]:  # check if units are Gt
+        nodata_check_bool = all(abs(i) < 10000 for i in change_values)
+
+    # If all rows passed the date check above, we store date_check_satisfied = True for this file: don't need to edit it
+    file_check_dataframe.loc[file_check_dataframe.local_filepath.__eq__(csv_file_path),
+                             'date_check_satisfied'] = date_check_bool
+
+    # Likewise if all rows passed the nodata check above, we store nodata_check_satisfied for this file.
+    file_check_dataframe.loc[file_check_dataframe.local_filepath.__eq__(csv_file_path),
+                             'nodata_check_satisfied'] = nodata_check_bool
+
+    return file_check_dataframe
+
+
+def edit_local_copies_of_glambie_csvs(file_check_dataframe: pd.DataFrame, local_path: str) -> pd.DataFrame:
```

Comment:
We know it's probably a glambie file. And we don't particularly care that the files are local. And we know the file check is a dataframe. I'd lean towards a function name like this:
```suggestion
def apply_csv_file_corrections(file_check_info: pd.DataFrame, directory_path: str) -> pd.DataFrame:
```

Reply:
Done